cmd/vault.go

```diff
@@ -397,3 +402,326 @@ func readJSON(name string) (secret interface{}, err error) {
+		Use:   "update",
+		Short: "Update user secret",
+		Long:  `Update existing user secret`,
+		Example: `
+	privx-cli user-secrets update [access flags] --owner-id <SECRET-NAME> --name <SECRET-NAME> JSON-FILE
+	privx-cli user-secrets update [access flags] --owner-id <SECRET-NAME> --name <SECRET-NAME>
+		--allow-read-to <ROLE-ID>
+		--allow-write-to <ROLE-ID>
+		...
+		JSON-FILE
+		`,
+		Args:         cobra.ExactArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return userSecretUpdate(options, args)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.secretName, "name", "", "secret name")
+	flags.StringArrayVar(&options.vaultReadTo, "allow-read-to", []string{}, "read by role ID")
+	flags.StringArrayVar(&options.vaultWriteTo, "allow-write-to", []string{}, "write by role ID")
+	flags.StringVar(&options.ownerID, "owner-id", "", "secret's owner ID")
+	cmd.MarkFlagRequired("owner-id")
+	cmd.MarkFlagRequired("name")
+
+	return cmd
+}
+
+func userSecretUpdate(options vaultOptions, args []string) error {
+	secret, err := readJSON(args[0])
+	if err != nil {
+		return err
+	}
+
+	secretID := &vault.SecretID{
+		OwnerID: options.ownerID,
+		Name:    options.secretName,
+	}
+	api := vault.New(curl())
+	bag, err := api.UserSecret(*secretID)
+	if err != nil {
+		return err
+	}
+
+	if len(options.vaultReadTo) == 0 {
+		for _, ref := range bag.AllowRead {
+			options.vaultReadTo = append(options.vaultReadTo, ref.ID)
+		}
+	}
+
+	if len(options.vaultWriteTo) == 0 {
+		for _, ref := range bag.AllowWrite {
+			options.vaultWriteTo = append(options.vaultWriteTo, ref.ID)
+		}
+	}
+
+	if err := api.UpdateUserSecret(*secretID, options.vaultReadTo,
+		options.vaultWriteTo, secret); err != nil {
+		return err
+	}
+
+	return stdout(secret)
+}
+
+//
+
+//show user secret metadata
+func userSecretMetadataShowCmd() *cobra.Command {
+	options := vaultOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "metadata",
+		Short: "Get a user secrets metadata",
+		Long:  `Get a user secrets metadata. Secret Name's are separated by commas when using multiple values, see example`,
+		Example: `
+	privx-cli user-secrets metadata [access flags] --owner-id <SECRET-NAME> --name <SECRET-NAME>,<SECRET-NAME>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return userSecretMetadataShow(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.secretName, "name", "", "secret name")
+	flags.StringVar(&options.ownerID, "owner-id", "", "secret's owner ID")
+	cmd.MarkFlagRequired("owner-id")
+	cmd.MarkFlagRequired("name")
+
+	return cmd
+}
+
+func userSecretMetadataShow(options vaultOptions) error {
+	api := vault.New(curl())
+	secrets := []vault.Secret{}
+
+	secretID := &vault.SecretID{
+		OwnerID: options.ownerID,
+		Name:    "",
+	}
+
+	for _, name := range strings.Split(options.secretName, ",") {
+
+		secretID.Name = name
+		secret, err := api.UserSecretMetadata(*secretID)
```

Comment:
the returned information is `metadata` or `secretMetadata` and is not a `secret`